homeassistant/components/media_player/firetv.py

```diff
@@ -93,13 +109,69 @@ def state(self):
     def action(self, action_id):
         """Perform an action on the device."""
         try:
-            requests.get(DEVICE_ACTION_URL.format(
-                self.host, self.port, self.device_id, action_id), timeout=10)
+            response = requests.get(DEVICE_ACTION_URL.format(
+                self.host, self.port, self.device_id, action_id),
+                timeout=10).json()
+            return response
         except requests.exceptions.RequestException:
             _LOGGER.error(
                 'Action request for %s was not accepted for device %s',
                 action_id, self.device_id)
 
+    def app(self, app_id, command):
+        try:
+            response = requests.get(
+                APPS_BASE_URL.format(self.host, self.port, self.device_id, app_id, command),
+                timeout=10).json()
+            _LOGGER.debug("response %s", response)
+            return response
+        except requests.exceptions.RequestException:
+            _LOGGER.error(
+                'Unable to execute %s for app %s',
+                command, app_id)
+
+
+class FireTVApp(MediaPlayerDevice):
+    def __init__(self, device, app, config):
+        self._device = device
+        self._name = app
+        self._app_id = config['app_id']
+        _LOGGER.warning("FOO: %s" % config)
```

Comment:
Decided that it's probably better to have it there in case someone wants to debug their config, fixed & changed to use debug().